include/flatbuffers/flexbuffers.h

```diff
@@ -743,7 +745,7 @@ inline Reference GetRoot(const uint8_t *buffer, size_t size) {
 }
 
 inline Reference GetRoot(const std::vector<uint8_t> &buffer) {
-  return GetRoot(buffer.data(), buffer.size());
+  return GetRoot(flatbuffers::vector_data(&buffer), buffer.size());
```

Comment:
Regarding

vector_data(vector* v) vs. vector_data(const vector& v)

there are mutable uses of vector.data()  Since it's possible for the result to be mutable, following Google style, it can't be a reference.  This makes it clear at the point of use that the data could be mutated.  The alternative would be to have

T* vector_data(vector<T>* v) 
T* vector_data(const vector& v)

which I feel is worse.  If you want me to go that way, happy to do so.